discover.go

```diff
@@ -24,5 +21,27 @@ func Discover(glob, dir string) ([]string, error) {
 		}
 	}
 
-	return filepath.Glob(filepath.Join(dir, glob))
+	ls, err := filepath.Glob(filepath.Join(dir, glob))
+	if err != nil {
+		return nil, err
+	}
+	var plugins []string
+
+	// Check for valid plugins files in glob matches
+	for _, f := range ls {
+		stats, err := os.Stat(f)
+		if err != nil {
+			return nil, err
```

Comment:
It seems like, if we can't stat the file, we should log a warning and continue. I could see how a failed stat could break an existing use case where those files would just be ignored by the plugin loader. I think we should log and continue.